ouroboros-consensus/test-crypto/Test/Crypto/DSIGN.hs

```diff
@@ -73,4 +74,4 @@ prop_dsign_verify_neg_msg :: forall a v. (Serialise a, Eq a, DSIGNAlgorithm v)
 prop_dsign_verify_neg_msg seed a a' sk = a /= a' ==>
     let sig = withSeed seed $ signDSIGN encode a sk
         vk  = deriveVerKeyDSIGN sk
-    in  not $ verifyDSIGN encode vk a' sig
+    in  verifyDSIGN encode vk a' sig =/= Right ()
```

Comment:
This is actually kind of nice; perhaps we should leave in the error string..? Or alternative, introduce it as a general wrapper in the class.